src/main/java/co/cask/spanner/common/SpannerConfig.java

```diff
@@ -0,0 +1,89 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.spanner.common;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.hydrator.common.ReferencePluginConfig;
+
+import java.io.IOException;
+import javax.annotation.Nullable;
+
+/**
+ * Spanner configuration
+ */
+public class SpannerConfig extends ReferencePluginConfig {
+  @Description("Google Cloud Project ID, which uniquely identifies your project." +
+    "You can find it on the Dashboard in the Cloud Platform Console")
+  @Macro
+  public String project;
+
+  @Description("Cloud Spanner instance id. " +
+    "Uniquely identifies Cloud Spanner instance within your Google Cloud Platform project")
+  @Macro
+  public String instance;
+
+  @Description("Cloud Spanner database id. Uniquely identifies your database within the Cloud Spanner instance")
+  @Macro
+  public String database;
+
+  @Name("serviceFilePath")
+  @Description("Path on the local file system to the service account key used for access to Spanner. " +
+    "Does not need to be specified when running on a Dataproc cluster. " +
+    "When running on other clusters, the file must be present on every node in the cluster.")
+  @Macro
+  @Nullable
+  public String serviceAccountFilePath;
+
+  @Name("schema")
+  @Description("Schema of the Spanner table")
+  @Macro
+  public String schema;
+
+  public SpannerConfig(String referenceName) {
+    super(referenceName);
+  }
+
+  public Schema getSchema() {
+    try {
+      return Schema.parseJson(schema);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Unable to parse output schema.");
+    }
+  }
+
+  protected void validate() {
+    if (!containsMacro("project") && project == null) {
+      throw new IllegalArgumentException("Project cannot be empty, " +
+                                           "please provide GCP project to use for Spanner");
+    }
+    if (!containsMacro("instance") && instance == null) {
+      throw new IllegalArgumentException("Cloud Spanner instance cannot be empty, " +
+                                           "please provide the Cloud Spanner instance in your GCP project");
+    }
+    if (!containsMacro("database") && database == null) {
+      throw new IllegalArgumentException("Cloud Spanner database cannot be empty, " +
+                                           "please provide the Cloud Spanner database in the Cloud Spanner instance");
+    }
+    if (!containsMacro("schema") && schema == null) {
+      throw new IllegalArgumentException("Schema cannot be empty, " +
+                                           "please provide the output schema for Cloud Spanner");
+    }
```

Comment:
@albertshau i added this validation check, this will be useful if a property has macro enabled but at runtime value is not provided, however its not a common scenario and we don't have these kind of checks in other plugins. So i am not sure if we need to keep this validation logic or remove it.